### ISSUE TITLE
Small change to avoid updating links with each Gmsh release

### ISSUE THREAD

**Issue by geuzaine** [mrkwjc/gmsh-sdk#10]
I've updated the continuous integration/delivery pipelines of Gmsh to automatically create symlinks to the latest stable versions. This means that the `gmsh-sdk` pip package could always reference the same package names, namely

 * http://gmsh.info/bin/Linux/gmsh-stable-Linux64-sdk.tgz
 * http://gmsh.info/bin/Linux/gmsh-stable-Linux32-sdk.tgz
 * http://gmsh.info/bin/Windows/gmsh-stable-Windows64-sdk.zip
 * http://gmsh.info/bin/Windows/gmsh-stable-Windows32-sdk.zip
 * http://gmsh.info/bin/MacOSX/gmsh-stable-MacOSX-sdk.tgz

What do you think?

As a side note, we can now reproduce the "downloads slow down at 99%" problem that some users have been reporting. It's due to the firewall behind which our server sits. We have reported the problem and are waiting for a fix.

**Comment by mrkwjc** [comment]
This will not simplify things very much because the platform have to be determined every time anyway and the url have to be composed using it. The only change will be that in setup.py the line:

```name = "gmsh-{}-{}-sdk".format(version, plat)```

will turn into:

```name = gmsh-{}-{}-sdk".format('stable', plat)```

But maybe this is a good idea to create new packages, i.e.: `gmsh-sdk-stable` and `gmsh-sdk-git` which will install always latest stable and git versions and there will be no need to change `version`  with every gmsh release. The problem is that such packages still have to have the version numbers. I saw that people often use something like 99999 for pakages like this.

BTW: I suspected the server side issue with downloads. Currently i switched to `requests` for downloads and people stopped complaining.

**Comment by geuzaine** [comment]
> This will not simplify things very much because the platform have to be determined every time anyway and the url have to be composed using it. 

Yes. What I meant is that you will not need to track Gmsh releases anymore, as your gmsh-sdk code does not need to be updated with each release anymore. The stable packages will always point to the latest released stable packages: our CD simply automatically creates the symlink after each new stable version is released.

> But maybe this is a good idea to create new packages, i.e.: gmsh-sdk-stable and gmsh-sdk-git which will install always latest stable and git versions and there will be no need to change version with every gmsh release. 

Yes, that's exactly what we do.

> The problem is that such packages still have to have the version numbers. I saw that people often use something like 99999 for pakages like this.

Of course: "http://gmsh.info/bin/Linux/gmsh-stable-Linux64-sdk.tgz" is just a symbolic link to whatever is the stable version; currently "http://gmsh.info/bin/Linux/gmsh-4.5.5-Linux64-sdk.tgz".

**Comment by mrkwjc** [comment]
Hi!
I created at pypi a new package named `gmsh-sdk-git` which should install latest git shapshot: https://pypi.org/project/gmsh-sdk-git. I can do the same with `gmsh-sdk-stable`, but actually i prefer not to do that because of versioning issues. `gmsh-sdk-stable` installer would have to have a fixed version number not covering the sdk version being actually installed. I think, this is acceptable for development version (git installer has version 9999), but questionable for stable install.

So i prefer to update the current `gmsh-sdk` with every new SDK release. I could automate that thing, just to not check this manually, but in daily cron tab. This automation would be much easier if there will be a fixed location where the current gmsh version string resides. Is there somewhere a file with such a string?

Regards,
Marek

**Comment by geuzaine** [comment]
> gmsh-sdk-stable installer would have to have a fixed version number not covering the sdk version being actually installed.

Ok, I get it!

Another possibility would be to integrate your code directly into Gmsh. Since our release process is fully automated, we could then automatically push the pip modifications as well?

> Is there somewhere a file with such a string?

The "master definition" of the version is in the main CMakeLists.txt file:
```
set(GMSH_MAJOR_VERSION 4)
set(GMSH_MINOR_VERSION 6)
set(GMSH_PATCH_VERSION 0)
set(GMSH_EXTRA_VERSION "")
```

**Comment by mrkwjc** [comment]
> Another possibility would be to integrate your code directly into Gmsh. Since our release process is fully automated, we could then automatically push the pip modifications as well?

This would be the logical step forward. :) The possible trouble i see is that `gmsh-sdk` is just the small **post-release** downloader and installer. It relies on the existing gmsh release files which should be accessible during testing and pushing (and installing of course). But maybe this is not a problem?

How should we then start with this integration?

**Comment by geuzaine** [comment]
> The possible trouble i see is that gmsh-sdk is just the small post-release downloader and installer. It relies on the existing gmsh release files which should be accessible during testing and pushing (and installing of course). But maybe this is not a problem?

I think this should be ok, as we directly push our binaries to the website during CI. (The script is here: https://gitlab.onelab.info/gmsh/gmsh/-/blob/master/.gitlab-ci.yml)

I have just created an account on PyPi. There actually seems to be no `gmsh` package! Since `gmsh-sdk` installs both the Gmsh app and the SDK, I think it would be nicer to create a new pip package simply called `gmsh`.

> How should we then start with this integration?

We could integrate your code e.g. as a new directory in `gmsh/utils`, then simply modify our CI pipeline to upload a new version of the `gmsh` pip package every time we do a stable release?

I just read through the PyPi tuto and created a token: it all seems simple enough ;-)

**Comment by mrkwjc** [comment]
I remeber now, that i left `gmsh` name intentionally free just for the case, you'd like to use it :). So i think this is right thing to do. Let me know when you are ready with this integration and i will announce here depreciation notice. Let me know also if i can help with the migration...

**Comment by geuzaine** [comment]
Ok I gave it a try by uploading two new packages: `gmsh` and `gmsh-dev`. Can you give them a try? They seem to work fine on my machine.

**Comment by mrkwjc** [comment]
Both installs do work nicely. Maybe the post versioning in `gmsh` should be removed as well?

**Comment by geuzaine** [comment]
Yes, it will be removed in the next release.

Thanks again for your help with this!

**Comment by mrkwjc** [comment]
Ok. With next gmsh release i will annotate `gmsh-sdk` as depreciated...
